cmake/CMakeLists.txt

```diff
@@ -1404,6 +1404,7 @@ function(onnxruntime_add_include_to_target dst_target)
     foreach(src_target ${ARGN})
```

Comment:
I wonder if we can have a better name for "onnxruntime_add_include_to_target" now that it is doing more than just adding includes

Reply:
What is the implication for other files that were just includes? There were not added as interface sources.
Also Natvis files were not used in `target_include/compile/directories/definitions`.

I would vote for a separate function.

Reply:
Normally INTERFACE_SOURCES is empty. Except for natvis files, I don't see it has any other usage. By default source files won't be propagated in such a way. You have to explicitly write code like:
```
target_sources(
        onnxruntime_common
        INTERFACE $<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/external/${ABSEIL_NATVIS_FILE}>)
```
When people are doing this, they must be doing it on purpose. So we shouldn't get unwanted source files included in our targets. And, if there were some, they should have been in the VC project of onnxruntime.dll. You can take a quick look. This PR only affects the VC projects that generate static libraries. @edgchen1 was suggesting this function deserves a new name. I haven't found one. Welcome to let us know what you think.